### ISSUE TITLE
Sample Solution to JS Ex 1: JavaScript Show and Hide 

### ISSUE THREAD

**Issue by ebeshero** [ebeshero/DHClass-Hub#252]
As I've been marking JavaScript homework, I noticed a common problem, so I'm posting some code to help provide an example. In JavaScript Ex 1, we asked you to try to write JavaScript that would display something hidden in your HTML file. You'd hide it with CSS, using `display:none`. Most of you could write a function to show an element you'd hidden, but you didn't know how to make it disappear again on a second click. 

You could always use the classList toggle if you're working with an `@class` attribute, like we did in JavaScript Ex 2, but if you want to access **any** HTML element, you'll want to play with if() else() statements, and it's good to know how to write this code or add this to the JavaScript you wrote for Ex 1. It's also good to know how to do the show/hide work on MULTIPLE elements (like table cells or paragraphs) as opposed to just one element with an `@id` attribute, so I've prepared an example to show you this, as a Solution to JavaScript Ex 1

 I've taken a homework assignment one of you submitted and added the code you'd need to toggle something to show and then to hide, by testing its CSS properties. To view it, check out the following:

1) HTML file with functioning JS show/hide: http://htmlpreview.github.io/?https://github.com/ebeshero/DHClass-Hub/blob/master/Class-Examples/JavaScript/showHide_If()Else()/showHideSamples.html

2) Now, [look at the HTML code](https://github.com/ebeshero/DHClass-Hub/blob/master/Class-Examples/JavaScript/showHide_If()Else()/showHideSamples.html)
TLDR: Notice: there are two tables, coded like this:
````
 <table id="ECCO">
            <tr>
                <th>Author</th>
                <th>Title</th>
              <!--  more table header cells. . .-->
</tr>
<tr>
     <td>WP</td>
     <td>WP</td>
<!--more td (or table data) cells -->
</tr>
</table>
<table id="NCCO">
    <tr>....</tr>
    <tr>...</tr>
</table>
````
And notice, there are two buttons coded on the page:
````
 <button>Click To See "Eighteenth-Century Collections Online" Results</button>
        . . .
 <button>Click To See "Nineteenth-Century Collections Online" Results</button>
````
3) Now [check out the CSS file](https://raw.githubusercontent.com/ebeshero/DHClass-Hub/master/Class-Examples/JavaScript/showHide_If()Else()/stylesheet.css)

In particular, take a look at these lines, which control *what's hidden about the two tables* when the page loads and before we press any buttons:

````
table#ECCO
{
/*...various stuff like font color and size...*/ 
**display:none;**
}
/*This means the table with @id of "ECCO" is completely hidden when we load the page. */

td, th
/*Notice, this styles <td> as well as <th> elements anywhere in the document*/
{
border: 1px solid #484848;
text-align: left;
padding: 10px;
}

table#NCCO tr td
{**display:none**}
/*This selects the <td> descendants of the table with the @id="NCCO". 
It's the CSS way to step down through the tree from table to tr to td.  
And we set the display on the table cell descendants of the
NCCO table to none. That means when we open the file, 
the table header row is the only thing showing for the NCCO table. */
````

4) Now we're ready to [check out the JavaScript file](https://raw.githubusercontent.com/ebeshero/DHClass-Hub/master/Class-Examples/JavaScript/showHide_If()Else()/script.js), which we'll just quote in full here. The showTable() function identifies the ECCO table and toggles it to display and then hide again.
The showTableRows2() function reaches for the table cells inside the NCCO table, and toggles those to display and then hide again, so you can see how to make a toggle work over PLURAL elements with a for-loop. 
````
window.addEventListener('DOMContentLoaded',init,false);
function init() {
var buttons = document.getElementsByTagName("button");
buttons[0].addEventListener('click', showTable, false)
buttons[1].addEventListener('click', showTableRows2, false)
}

function showTable() {
var table1 = document.getElementById("ECCO")
if (table1.style.display == 'none') 
{table1.style.display='block';}
else {table1.style.display='none';}
} 
 /* You may be wondering why we don't code the "if" statement like this:
 if (table1.style.display = 'none'). 
**That  won't work** because '=' in JavaScript doesn't test for a condition of equality, rather it
 ASSIGNS a new value. What you need to do is test for a condition, and for that we need
the DOUBLE EQUAL SIGN: 
if (table1.style.display == 'none')
 */


function showTableRows2() {
var table2 = document.getElementById("NCCO")
var tCells = table2.getElementsByTagName("td")
for (var i = 0, length = tCells.length; i < length; i++)
if (tCells[i].style.display == 'none') 
{tCells[i].style.display='table-cell';}
else {tCells[i].style.display='none';}
}
/*getElementsByTagName works to look for all the 
descendant elements of the "NCCO" table named "td". 
Then we set up a for loop to step through
 each table cell in turn. */

````

Notice: we used a CSS display value here of "table-cell". We did that after trying the standard "block" and "inline" and seeing the output look pretty awful in the browser. 
Lesson: Table elements have special CSS display properties, which can be found here:
 http://www.w3schools.com/csSref/pr_class_display.asp .

**Comment by jonhoranic** [comment]
@ebeshero It was the double equal sign this **ENTIRE TIME** and I never noticed! Thank you for clearing that up! I think I can now start applying JavaScript to what I have done so far on Exercise 3. I will no doubt stalking the GitHub and asking questions on more complex pieces of JavaScript I am going to test out.

**Comment by RJP43** [comment]
Closing for Spring 2017 Semester
